extension/app/ts/components/formVisualizerResults.ts

```diff
@@ -42,8 +41,8 @@ export function formSimulatedAndVisualizedTransaction(simState: SimulationState,
 					token: tokenEntry
 				}
 			}
-			throw new Error('wrong token type')
-		})
+			return undefined // a token that is not NFT, but does not have decimals either, let's just not visualize them
+		}).filter((x): x is TokenVisualizerResultWithMetadata => x !== undefined)
```

Comment:
Syntax may be a bit off here, but this is a more accurate signature and will help protect against bugs by avoiding blind typecasting.
```suggestion
		}).filter(<T>(x: T | undefined): x is T => x !== undefined)
```